internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go

```diff
@@ -206,21 +224,47 @@ func updateFleetConfig(log *logger.Logger, src remote.Config, dst *remote.Config
 		urlCopy := *src.Transport.Proxy.URL
 		dst.Transport.Proxy.URL = &urlCopy
 
-		log.Debug("received proxy from fleet, applying it")
+	}
+
+	if src.Transport.TLS != nil {
+
+		if dst.Transport.TLS == nil {
+			dst.Transport.TLS = new(tlscommon.Config)
+		} else {
+			// copy the TLS struct
+			tlsCopy := *dst.Transport.TLS
+			dst.Transport.TLS = &tlsCopy
+		}
+
+		emptyCertificate := tlscommon.CertificateConfig{}
+		if src.Transport.TLS.Certificate == emptyCertificate {
+			log.Debug("TLS certificates from fleet are empty or null, the TLS config will not be changed")
+		} else {
+			dst.Transport.TLS.Certificate = src.Transport.TLS.Certificate
+			log.Debug("received TLS certificate/key from fleet, applying it")
+		}
+
+		if len(src.Transport.TLS.CAs) == 0 {
+			log.Debug("TLS CAs from fleet are empty or null, the TLS config will not be changed")
+		} else {
+			dst.Transport.TLS.CAs = make([]string, len(src.Transport.TLS.CAs))
+			copy(dst.Transport.TLS.CAs, src.Transport.TLS.CAs)
+			log.Debug("received TLS CAs from fleet, applying it")
+		}
 	}
 }
 
 func (h *PolicyChangeHandler) handlePolicyChange(ctx context.Context, c *config.Config) (err error) {
-	cfg, err := configuration.NewFromConfig(c)
-	if err != nil {
-		return errors.New(err, "could not parse the configuration from the policy", errors.TypeConfig)
-	}
+	//cfg, err := configuration.NewFromConfig(c)
+	//if err != nil {
+	//	return errors.New(err, "could not parse the configuration from the policy", errors.TypeConfig)
+	//}
```

Comment:
Just delete this now as the configuration is just passed into the `validateFleetServerHosts`